src/autopas/selectors/Configuration.h

```diff
@@ -114,13 +125,12 @@ inline bool operator!=(const Configuration& lhs, const Configuration& rhs) { ret
  * @return
  */
 inline bool operator<(const Configuration& lhs, const Configuration& rhs) {
-  return std::tie(lhs._container, lhs._traversal, lhs._dataLayout, lhs._newton3) <
-         std::tie(rhs._container, rhs._traversal, rhs._dataLayout, rhs._newton3);
+  return std::tie(lhs.container, lhs.traversal, lhs.dataLayout, lhs.newton3, lhs.cellSizeFactor) <
```

Comment:
We probably want the cellSizeFactor directly after container. This is because of this comparison defines the order the configurations are tested, at least in full search. You want configurations that need minimal rebuilding to be near each other. Changing the cellSizeFactor probably leads to rebuilding the container and is, therefore, costlier than changing the traversal.
Also reflect this in the ordering in `toString()` and the constructor